rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,535 @@
+
+        return result
+
+    def _paginated_api_query(
+            self,
+            endpoint: Literal[
+                'order/history',
+                'asset/deposit/query-record',
+                'asset/withdraw/query-record',
+            ],
+            options: dict[str, Any],
+    ) -> list[dict[str, Any]]:
+        if endpoint == 'order/history':
+            result_key = 'list'
+        else:
+            result_key = 'rows'
+
+        result = []
+        # copy since we are modifying the dict and could affect other queries using
+        # the same options
+        query_options = options.copy()
+        while True:
+            output = self._api_query(path=endpoint, options=query_options)
+            if (next_cursor := output.get('nextPageCursor')) is not None and len(next_cursor) != 0:
+                query_options['cursor'] = next_cursor
+
+            result.extend(output[result_key])
+            if len(output[result_key]) < query_options.get(result_key, PAGINATION_LIMIT):
+                break
+
+        return result
+
+    def query_online_trade_history(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> tuple[list[Trade], tuple[Timestamp, Timestamp]]:
+        """
+        Query trades from bybit in the spot category.
+        The API is limited to query trades up to two years into the past. We can query time ranges
+        but we don't know when to stop querying into the past. What this function does is iterate
+        over the 2 years period moving the queried frame in the maximum allowed time span, that is
+        one week:
+        [end ts, end ts - 1 week] -> [end ts - 1w, end ts - 2w] -> ... [end ts - n*w, start ts]
+        """
+        new_trades = []
+        upper_ts, lower_ts = end_ts, Timestamp(end_ts - WEEK_IN_SECONDS)
+        two_years_ago = Timestamp(ts_now() - DAY_IN_SECONDS * 365 * 2)
+        if start_ts < two_years_ago:
+            # bybit allows to query up to two years into the past
+            start_ts = two_years_ago
+
+        while True:
+            raw_data = self._paginated_api_query(
+                endpoint='order/history',
+                options={
+                    'category': 'spot',
+                    'endTime': str(ts_sec_to_ms(upper_ts)),
+                    'limit': PAGINATION_LIMIT,
+                    'startTime': str(ts_sec_to_ms(lower_ts)),
+                },
+            )
+            for raw_trade in raw_data:
+                if (order_status := raw_trade.get('orderStatus')) != 'Filled':
+                    log.debug(f'Skipping entry {raw_trade} with status {order_status}')
+                    continue  # api doesn't allow to filter by status in the classic spot
+
+                try:
+                    base_asset, quote_asset = bybit_symbol_to_base_quote(raw_trade['symbol'])
+                except (UnknownAsset, UnprocessableTradePair, KeyError):
```

Comment:
What are you doing here? You handle key error by ignoring it and creating another key error? You are trying to access `raw_trade["symbol"]` in the exception handler again